fix fmi2 headers to avoid warnings about redefinition of symbols

The headers of the FMI2 library redefine symbols which gives the warning below. This issue is to avoid redefining the symbols.
```
$ jm_ipython.sh jmodelica.py Buildings.ThermalZones.EnergyPlus.Validation.ThreeZonesTwoBuildings
WARNING:root:IPOPT_HOME="": Path does not exist. An IPOPT installation could not be found, some modules and examples will therefore not work properly.
*** Compiling Buildings.ThermalZones.EnergyPlus.Validation.ThreeZonesTwoBuildings
sources/ModelicaInternal_fullPathName.c: In function ‘main’:
sources/ModelicaInternal_fullPathName.c:56:24: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
             fullName_v = ModelicaInternal_fullPathName(name_v);
                        ^
/opt/oct/lib/RuntimeLibrary/libModelicaExternalC.a(ModelicaInternal.c.o): In function `ModelicaInternal_temporaryFileName':
/opt/OCT/P538-OCT/JModelica/ThirdParty/MSL/Modelica/Resources/C-Sources/ModelicaInternal.c:657: warning: the use of `tmpnam' is dangerous, better use `mkstemp'
Warning at line 138, column 3, in file '/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Fluid/Interfaces/ConservationEquation.mo',
In components:
    zon1.zon
    zon2.zon1.zon
    zon2.zon2.zon
  Evaluate annotation is ignored for parameters with fixed=false
Warning at line 49, column 12, in file '/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Fluid/Sources/BaseClasses/PartialSource.mo',
In components:
    zon1.bou
    zon1.freshAir
    zon2.zon1.bou
    zon2.zon1.freshAir
    .. and 2 additional components
  The cardinality() function-like operator is deprecated, and will be removed in a future version of Modelica
sources/ModelicaInternal_readLine.c: In function ‘main’:
sources/ModelicaInternal_readLine.c:64:22: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
             string_v = ModelicaInternal_readLine(fileName_v, tmp_1, &tmp_2);
                      ^
/opt/oct/lib/RuntimeLibrary/libModelicaExternalC.a(ModelicaInternal.c.o): In function `ModelicaInternal_temporaryFileName':
/opt/OCT/P538-OCT/JModelica/ThirdParty/MSL/Modelica/Resources/C-Sources/ModelicaInternal.c:657: warning: the use of `tmpnam' is dangerous, better use `mkstemp'
In file included from /home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI/fmi_import_context.h:31:0,
                 from /home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/fmilib.h:27,
                 from /home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/EnergyPlusTypes.h:10,
                 from /home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/EnergyPlusFMU.h:8,
                 from /home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/ZoneFree.h:9,
                 from /home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/ZoneFree.c:8,
                 from sources/Buildings_ThermalZones_EnergyPlus_Validation_ThreeZonesTwoBuildings_funcs.c:2:
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2_types.h:37:34: warning: redefinition of typedef ‘fmi2_component_environment_t’ [-Wpedantic]
 #define fmi2ComponentEnvironment fmi2_component_environment_t
                                  ^
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2TypesPlatform.h:100:28: note: in expansion of macro ‘fmi2ComponentEnvironment’
    typedef void*           fmi2ComponentEnvironment;    /* Pointer to FMU environment    */
                            ^~~~~~~~~~~~~~~~~~~~~~~~
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2_types.h:37:34: note: previous declaration of ‘fmi2_component_environment_t’ was here
 #define fmi2ComponentEnvironment fmi2_component_environment_t
                                  ^
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2TypesPlatform.h:100:28: note: in expansion of macro ‘fmi2ComponentEnvironment’
    typedef void*           fmi2ComponentEnvironment;    /* Pointer to FMU environment    */
                            ^~~~~~~~~~~~~~~~~~~~~~~~
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2_types.h:38:22: warning: redefinition of typedef ‘fmi2_FMU_state_t’ [-Wpedantic]
 #define fmi2FMUstate fmi2_FMU_state_t
                      ^
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2TypesPlatform.h:101:28: note: in expansion of macro ‘fmi2FMUstate’
    typedef void*           fmi2FMUstate;                /* Pointer to internal FMU state */
                            ^~~~~~~~~~~~
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2_types.h:38:22: note: previous declaration of ‘fmi2_FMU_state_t’ was here
 #define fmi2FMUstate fmi2_FMU_state_t
                      ^
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2TypesPlatform.h:101:28: note: in expansion of macro ‘fmi2FMUstate’
    typedef void*           fmi2FMUstate;                /* Pointer to internal FMU state */
                            ^~~~~~~~~~~~
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2_types.h:43:18: warning: redefinition of typedef ‘fmi2_char_t’ [-Wpedantic]
 #define fmi2Char fmi2_char_t
                  ^
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2TypesPlatform.h:106:28: note: in expansion of macro ‘fmi2Char’
    typedef char            fmi2Char;
                            ^~~~~~~~
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2_types.h:43:18: note: previous declaration of ‘fmi2_char_t’ was here
 #define fmi2Char fmi2_char_t
                  ^
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2TypesPlatform.h:106:28: note: in expansion of macro ‘fmi2Char’
    typedef char            fmi2Char;
                            ^~~~~~~~
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2_types.h:45:18: warning: redefinition of typedef ‘fmi2_byte_t’ [-Wpedantic]
 #define fmi2Byte fmi2_byte_t
                  ^
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2TypesPlatform.h:108:28: note: in expansion of macro ‘fmi2Byte’
    typedef char            fmi2Byte;
                            ^~~~~~~~
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2_types.h:45:18: note: previous declaration of ‘fmi2_byte_t’ was here
 #define fmi2Byte fmi2_byte_t
                  ^
/home/mwetter/proj/ldrd/bie/modeling/github/lbl-srg/modelica-buildings/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2TypesPlatform.h:108:28: note: in expansion of macro ‘fmi2Byte’
    typedef char            fmi2Byte;
                            ^~~~~~~~

```

--- a/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2TypesPlatform.h
+++ b/Buildings/Resources/C-Sources/EnergyPlus/FMI2/fmi2TypesPlatform.h
@@ -38,7 +38,7 @@
                      Hans Olsson (Dynasim).
 
 
-   Copyright © 2008-2011 MODELISAR consortium,
+   Copyright ï¿½ 2008-2011 MODELISAR consortium,
                2012-2013 Modelica Association Project "FMI"
                All rights reserved.
    This file is licensed by the copyright holders under the BSD 2-Clause License
@@ -97,15 +97,27 @@
    fmi2Byte                : smallest addressable unit of the machine, typically one byte.
 */
    typedef void*           fmi2Component;               /* Pointer to FMU instance       */
+#ifndef fmi2Component_defined
+#define fmi2Component_defined
    typedef void*           fmi2ComponentEnvironment;    /* Pointer to FMU environment    */
+#endif
+#ifndef fmi2FMUstate_defined
+#define fmi2FMUstate_defined
    typedef void*           fmi2FMUstate;                /* Pointer to internal FMU state */
+#endif
    typedef unsigned int    fmi2ValueReference;
    typedef double          fmi2Real   ;
    typedef int             fmi2Integer;
    typedef int             fmi2Boolean;
+#ifndef fmi2Char_defined
+#define fmi2Char_defined
    typedef char            fmi2Char;
+#endif
    typedef const fmi2Char* fmi2String;
+#ifndef fmi2Byte_defined
+#define fmi2Byte_defined
    typedef char            fmi2Byte;
+#endif
 
 /* Values for fmi2Boolean  */
 #define fmi2True  1

